content/en/docs/ops/common-problems/network-issues/index.md

```diff
@@ -223,6 +223,79 @@ server {
 }
 {{< /text >}}
 
+## Virtual service with fault injection and retry/timeout policies not working as expected
+
+Currently, Istio does not support configuring fault injections and retry or timeout policies on the
+same `VirtualService`. Consider the following configuration:
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: helloworld
+spec:
+  hosts:
+    - "*"
+    gateways:
+    - helloworld-gateway
+    http:
+    - match:
+      - uri:
+          exact: /hello
+        fault:
+          abort:
+            httpStatus: 500
+            percentage:
+              value: 50
+        retries:
+          attempts: 5
+          retryOn: 5xx
+        route:
+        - destination:
+          host: helloworld
+          port:
+          number: 5000
+{{< /text >}}
+
+You would expect that given the configured five retry attempts, the user would almost never see any
+errors when calling the `helloworld` service. However since both fault and retries are configured on
+the same `VirtualService`, the retry configuration does not take effect, resulting in a 50% failure
+rate. To work around this issue, you may remove the fault config from your `VirtualService` and
+inject the fault to the upstream Envoy proxy using `EnvoyFilter` instead:
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: EnvoyFilter
+metadata:
+  name: hello-world-filter
+spec:
+  workloadSelector:
+    labels:
+      app: helloworld
+  configPatches:
+    - applyTo: HTTP_FILTER
+      match:
+        context: SIDECAR_INBOUND # will match outbound listeners in all sidecars
+        listener:
+          filterChain:
+            filter:
+              name: "envoy.filters.network.http_connection_manager"
+      patch:
+        operation: INSERT_BEFORE
+        value:
+          name: envoy.fault
+          typed_config:
+            "@type": "type.googleapis.com/envoy.extensions.filters.http.fault.v3.HTTPFault"
+            abort:
+              http_status: 500
+              percentage:
+                numerator: 50
+                denominator: HUNDRED
+{{< /text >}}
+
+This works because this way the retry policy is configured for the client proxy while the fault
+injection is configured for the upstream proxy.
+
```

Comment:
Not sure, but it might be worth mentioning the alternative of moving the retry policy to the virtual service of a client service, if that makes sense? For example, in bookinfo, it would probably be reasonable to put the retry in the `reviews` service, for fault injection in the `ratings` service.

Reply:
That is true. I didn't mention this specifically because configuring retries and faults on _different_ virtual services is quite the obvious workaround. I'll see if I can come up with a sentence or two that mentions this as well.